client/client.go

```diff
@@ -323,16 +335,26 @@ func (c *Client) NewUnAuthUploadRequest(ctx context.Context, method, urlStr stri
 
 	u := c.BaseURL.ResolveReference(rel)
 
-	buf := bytes.NewBuffer(body)
+	req, err := http.NewRequest(method, u.String(), fileReader)
 
-	req, err := http.NewRequest(method, u.String(), buf)
+	if err != nil {
+		return nil, err
+	}
 
+	// Set req.ContentLength and req.GetBody as internally there is no implementation of such for os.File type reader
+	// http.NewRequest() only sets this values for types - bytes.Buffer, bytes.Reader and strings.Reader
+	// Refer https://github.com/golang/go/blob/a0f77e56b7a7ecb92dca3e2afdd56ee773c2cb07/src/net/http/request.go#L896
```

Comment:
Nit: While this is true for those data types, it doesn't seem necessary when using an *os.File, see my example in the issue -- it uploaded perfectly fine without pre-emptively specifying content-length. That being said, I totally support not relying on default behavior here.